contributors/design-proposals/daemonset-update.md

```diff
@@ -165,22 +187,38 @@ DaemonSets on the apiserver.
 
 For each pending DaemonSet updates, it will:
 
-1. Find all pods whose labels are matched by DaemonSet `.spec.selector`. 
-   - If `OwnerReference` is implemented for DaemonSets, filter out pods that
-     aren't controlled by this DaemonSet too.
+1. Find all pods controlled by DaemonSet
 1. Check `DaemonSetUpdateStrategy`:
    - If `OnDelete`: do nothing
    - If `RollingUpdate`:
-     - Find all daemon pods that belong to this DaemonSet using label selectors.
-       - Pods with "pod-template-generation" value equal to the DaemonSet's
-         `.spec.templateGeneration` are new pods, otherwise they're old pods.
-       - Note that pods without "pod-template-generation" labels (e.g. DaemonSet
-         pods created before RollingUpdate strategy is implemented) will be
-         seen as old pods.
+     - Find existing PodTemplates owned by DaemonSet, and sort them by the value
+       of `pod-template-generation` label
+     - Clean up PodTemplates based on `.spec.revisionHistoryLimit`
```

Comment:
Now if I want to apply multiple updates to a DaemonSet template w/o automatically rolling it out (use a flow like the following), I'll have to use OnDelete and then switch back to Rolling thus use the switch as a trigger to rollout all of my updates in a single unit.
```sh
# using OnDelete
# updated image
kubectl set image ds foo foo-sidecar=sidecar:0.2
# updated config
kubectl set volume --configmap-name=foo-cm-fa4dbh --overwrite
# maybe configure a missing probe
kubectl set probe --readiness ...
# switch back to Rolling
```
I guess what I am arguing for is keeping this consistent with Deployments and not have two different ways to do the same thing.